### PR TITLE
[buteo-sync-plugins-social] Set summary+description info into Google Cal...

### PR DESCRIPTION
**Pull request by chriadam**
...endars

Previously, we hardcoded calendar summary information for calendars
synced from Google.  Instead, we should use the summary and description
provided by the server.

### CODE REVIEW

**Comment by chriadam** [comment]
Requires https://github.com/nemomobile/libsocialcache/pull/58 because I broke it with a recent fix to caldav.

**Comment by chriadam** [comment]
@pvuorela @pgerdt

File: src/google/google-calendars/googlecalendarsyncadaptor.h [hunk 1]
```diff
@@ -77,7 +77,13 @@ private Q_SLOTS:
     void upsyncFinishedHandler();
 
 private:
-    QMap<int, QMap<QString, QPair<QString, QString> > > m_serverCalendarIdToSummaryAndColor;
+    struct CalendarInfo {
+        QString summary;
+        QString description;
+        QString color;
+        QString etag;
```
Comment (pgerdt) [inline src/google/google-calendars/googlecalendarsyncadaptor.h]: CalendarInfo::etag is only assigned to, not read. I guess it is reserved for future use?

**Comment by pgerdt** [comment]
Tested together with https://github.com/nemomobile/nemo-qml-plugin-calendar/pull/51, Google cal description comes through and updates when the remote changes.

File: src/google/google-calendars/googlecalendarsyncadaptor.h (hunk 1 above)
Comment (chriadam) [inline src/google/google-calendars/googlecalendarsyncadaptor.h]: I have now removed it again, as we probably won't use it in the future.  If we do, we can add it then.

**Comment by pgerdt** [comment]
LGTM

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**